synapse/config/server.py

```diff
@@ -302,6 +302,24 @@ def read_config(self, config: JsonDict, **kwargs: Any) -> None:
                 "public_baseurl cannot contain query parameters or a #-fragment"
             )
 
+        self.extra_well_known_content = config.get("extra_well_known_content")
+
+        if type(self.extra_well_known_content) is not dict:
```

Comment:
This is fine, but we'd usually write this as

```suggestion
        if not isinstance(self.extra_well_known_content, dict):
```

No real benefit here other than consistency across Synapse.